extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -250,8 +250,10 @@ protected TaskLocation getTaskLocation()
           podStatus.getPodIP(),
           DruidK8sConstants.PORT,
           DruidK8sConstants.TLS_PORT,
-          Boolean.parseBoolean(pod.getMetadata().getAnnotations().getOrDefault(DruidK8sConstants.TLS_ENABLED, "false"))
+          Boolean.parseBoolean(pod.getMetadata().getAnnotations().getOrDefault(DruidK8sConstants.TLS_ENABLED, "false")),
+          pod.getMetadata() != null ? pod.getMetadata().getName() : ""
       );
+      log.info(StringUtils.format("K8s task %s is running at location %s", taskId.getOriginalTaskId(), taskLocation));
```

Comment:
It is better not to use StringUtils.format in the log message directly as this will cause the string to be formatted even if the log level is set to a higher level like `warn`

```suggestion
      log.info("K8s task %s is running at location %s", taskId.getOriginalTaskId(), taskLocation);
```

Reply:
we will not have the correct pod name in the constructor